internal/providers/pke/pkeworkflow/create_cluster.go

```diff
@@ -198,26 +255,37 @@ func (w CreateClusterWorkflow) Execute(ctx workflow.Context, input CreateCluster
 		return errors.Errorf("couldn't get the default security group of the VPC %q", vpcOutput["VpcId"])
 	}
 
-	var nodePools []NodePool
-
-	// List node pools
+	// Create subnets map nodepools with subnet?
+	// We need a Zone - SubnetID map
+	subnetIDMap := make(map[string]string)
 	{
-		activityInput := ListNodePoolsActivityInput{ClusterID: input.ClusterID}
-		err := workflow.ExecuteActivity(ctx, ListNodePoolsActivityName, activityInput).Get(ctx, &nodePools)
-		if err != nil {
-			return err
+		var createSubnetFutures []workflow.Future
+		for zone, ip := range availabilityZoneMap {
+			activityInput := CreateSubnetActivityInput{
+
+				AWSActivityInput: awsActivityInput,
+				ClusterID:        input.ClusterID,
+				ClusterName:      input.ClusterName,
+				VpcID:            vpcOutput["VpcId"],
+				RouteTableID:     vpcOutput["RouteTableId"],
+				Cidr:             ip,
+				AvailabilityZone: zone,
+			}
+			ctx := workflow.WithActivityOptions(ctx, aoWithHeartbeat)
```

Comment:
`CreateSubnetActivityName` is started with a Cadence `HeartBeatTimeout` but there is no heart beating in that activity.